ocean_lib/ocean/ocean.py

```diff
@@ -335,33 +334,23 @@ def create_pool(
         return bpool
 
     @enforce_types
-    def build_compute_provider_fees(
-        self,
-        provider_data: Union[str, dict],
-        provider_fee_address: str,
-        provider_fee_token: str,
-        provider_fee_amount: int,
-        valid_until: int,
-    ) -> tuple:
-        if isinstance(provider_data, dict):
-            provider_data = json.dumps(provider_data, separators=(",", ":"))
-
-        message = self.web3.solidityKeccak(
-            ["bytes", "address", "address", "uint256", "uint256"],
-            [
-                self.web3.toHex(self.web3.toBytes(text=provider_data)),
-                provider_fee_address,
-                provider_fee_token,
-                provider_fee_amount,
-                valid_until,
-            ],
+    def retrieve_provider_fees(self, asset: Asset, publisher_wallet: Wallet) -> tuple:
+        data_provider = DataServiceProvider
```

Comment:
I think there is some data provider already set on ocean, isn't there?
Also, regarding the next line, I think should we pass in the service as well, as a parameter to this function. If there is no access service, the error here will be ugly.

Reply:
But since we use `ocean.assets.create` function before, in order to create an asset, you have already an access service generated by default.